java-checks/src/main/java/org/sonar/java/checks/StandardCharsetsConstantsCheck.java

```diff
@@ -227,13 +234,35 @@ public void visitNode(Tree tree) {
     }
   }
 
+  List<JavaQuickFix> quickFixesOnMemberSelect(IdentifierTree identifierTree) {
+    Tree parent = identifierTree.parent();
+    if (parent.is(Tree.Kind.MEMBER_SELECT)) {
+      MemberSelectExpressionTree parentMemberSelect = (MemberSelectExpressionTree) parent;
+      return List.of(
+        JavaQuickFix.newQuickFix(REPLACE_WITH_STANDARD_CHARSETS + identifierTree.name() + "\"")
+          .addTextEdit(JavaTextEdit.replaceTree(parentMemberSelect.expression(), "java.nio.charset.StandardCharsets"))
+          .build()
+      );
+    }
+    return Collections.emptyList();
+  }
+
+  private void reportQuickfixOnMemberSelect(IdentifierTree identifierTree, String identifier) {
+    QuickFixHelper.newIssue(context)
+      .forRule(this)
+      .onTree(identifierTree)
+      .withMessage("Replace \"com.google.common.base.Charsets." + identifier + "\" with \"StandardCharsets." + identifier + "\".")
+      .withQuickFixes(() -> quickFixesOnMemberSelect(identifierTree))
+      .report();
+  }
+
```

Comment:
How about moving this block below the place with the newer additional methods that generate quickfixes?